skademlia/protocol_test.go

```diff
@@ -42,3 +44,20 @@ func TestAdressMatches(t *testing.T) {
 		assert.True(t, addressMatches(hostMatchTest.bind, hostMatchTest.subject) == hostMatchTest.equal, hostMatchTest.bind+" compared to "+hostMatchTest.subject)
 	}
 }
+
+func TestQuickCheckAddressMatches(t *testing.T) {
+	f := func(n int) bool {
```

Comment:
To make the addresses more random, I'd have `func(a, b, c, d byte, port uint16) bool` and stitch up `a, b, c, d` into an IP address via `strconv.FormatUint`.

An alternative option by the way is that `net.IP` is just a byte slice. Can randomly generate bytes of IPv4 or IPv6 length, type-cast them into a `net.IP` instance to see how feeding arbitrary bytes might potentially cause some edge cases for `addressMatches`.